src/components/MainParticipant/MainParticipant.tsx

```diff
@@ -2,14 +2,19 @@ import MainParticipantInfo from '../MainParticipantInfo/MainParticipantInfo';
 import ParticipantTracks from '../ParticipantTracks/ParticipantTracks';
 import React from 'react';
 import useMainSpeaker from '../../hooks/useMainSpeaker/useMainSpeaker';
+import useSelectedParticipant from '../VideoProvider/useSelectedParticipant/useSelectedParticipant';
 
 export default function MainParticipant() {
   const mainParticipant = useMainSpeaker();
+  const [selectedParticipant] = useSelectedParticipant();
+
+  const videoPriority = mainParticipant === selectedParticipant ? 'high' : null;
+
   return (
     /* audio is disabled for this participant component because this participant's audio 
        is already being rendered in the <ParticipantStrip /> component.  */
     <MainParticipantInfo participant={mainParticipant}>
-      <ParticipantTracks participant={mainParticipant} disableAudio enableScreenShare videoPriority="high" />
+      <ParticipantTracks participant={mainParticipant} disableAudio enableScreenShare videoPriority={videoPriority} />
```

Comment:
Can I understand the rationale behind this line? Does this mean that the dominant speaker is assigned 'high' priority once it's rendered in the focus area (i.e. the main area of the screen) after the dominant-speaker event is received? If that's the case, we should not assign high-priority to the dominant speaker. The dominant-speaker priority should be only managed by VMS.

Reply:
Here, `videoPriority` is set to 'high' only when the user pins a participants thumbnail. When the dominant speaker is to be rendered in the main video area, `videoPriority` will be `null` and `track.setPriority` will never be called. 

I suppose there could be a case when the user pins the dominant speaker.  In this case `track.setPriority('high')` _would_ be called on the dominant speaker's video track.  I feel like this would be an uncommon occurrence, but I can add some logic to avoid this if you think it would be worthwhile.

Reply:
@timmydoza thanks for the explanation. You can keep it as it is, no need to change anything. Just FYI, dominantSpeakerPriority only upgrades the priority, never downgrades it. In other words, if the user pins a video and that video becomes dominant speaker, the priority will stay as 'high' even if dominantSpeakerPriority='standard'. Hence, IMO your logic works right.

Reply:
I see.  I did update the logic so that `setPriority` is never called on a dominant speaker track.  Do you think I should revert that so a pinned dominant speaker will have high priority instead of standard?

Reply:
I agree that it would be simpler to always have pinned video be high priority.  I'll update the code accordingly.